imports/client/TestRunner.js

```diff
@@ -0,0 +1,190 @@
+            () => {return true;},
+            (minute) => {
+                if (!minute.topics || minute.topics.length === 0)
+                    return 'This minute has no topics';
+            }
+        ));
+
+        // no participant marked as present (F)
+        TestCase.testCases.push(new TestCase('No participants marked as present',
+            TestRunner.TRIGGERS.finalize,
+            () => {return true;},
+            (minute) => {
+                let noParticipantsPresent = true;
+                minute.participants.forEach(p => {
+                    if(p.present) noParticipantsPresent = false;
+                    }
+                );
+                if(noParticipantsPresent)
+                    return 'No participant is marked as present'
+            }
+        ));
+
+        // no topics checked (F)
+        TestCase.testCases.push(new TestCase('No topic is checked',
+            TestRunner.TRIGGERS.finalize,
+            () => {return true;},
+            (minute) => {
+                if(minute.topics.length < 0) return;
+                let noTopicChecked = true;
+                minute.topics.forEach(topic => {
+                        if(!topic.isOpen) noTopicChecked = false;
+                    }
+                );
+                if(noTopicChecked)
+                    return 'No topic is checked'
+            }
+        ));
+
+        // topic checked but no children (F)
+        TestCase.testCases.push(new TestCase('A topic is checked but has no children',
+            TestRunner.TRIGGERS.finalize,
+            () => {return true;},
+            (minute) => {
+                if(minute.topics.length < 0) return;
+
+                let checkedButChildren = false;
+                minute.topics.forEach(topic => {
+                        if(topic.isOpen) return;
+                        if(!topic.infoItems || topic.infoItems.length === 0) checkedButChildren = true;
+                    }
+                );
+                if(checkedButChildren)
+                    return 'A topic is checked but has no children'
+            }
+        ));
+
+        // action item with no responsible (F)
+        TestCase.testCases.push(new TestCase('Action item has no responsible',
+            TestRunner.TRIGGERS.finalize,
+            () => {return true;},
+            (minute) => {
+                if(minute.topics.length < 0) return;
+
+                let actionItemWithoutResponsible = false;
+                minute.topics.forEach(topic => {
+                    topic.infoItems.forEach(infoItem => {
+                            if(infoItem.itemType === "actionItem" && ((!infoItem.responsibles) || (infoItem.responsibles.length === 0)))
+                                actionItemWithoutResponsible = true;
+                        });
+                });
+                if(actionItemWithoutResponsible)
+                    return 'An action item has no responsible'
```

Comment:
In case we habe 30 topics and each topic 3 action items. Which of the 90 action items has a problem? Could you please investigate if its possible to add the e.g., first 20 chars of the action item title to the warning message box?